crates/burn-import/src/onnx/op_configuration.rs

```diff
@@ -1015,6 +1015,53 @@ pub fn shape_config(curr: &Node) -> (usize, usize) {
     (start_dim as usize, end_dim as usize)
 }
 
+pub fn slice_config(
+    node: &Node,
+) -> (
+    Vec<usize>,
+    Vec<usize>, /*, Option<Vec<i64>>, Option<Vec<i64>> */
+) {
+    let start_value = &node.inputs[1].value;
+    let end_value = &node.inputs[2].value;
+
+    let starts = match &node.inputs[1].ty {
+        ArgType::Tensor(tensor) => {
+            assert_eq!(tensor.dim, 1, "Slice: ends tensor must be 1D");
+            if let Some(Data::Int64s(shape)) = start_value.as_ref() {
+                shape.iter().map(|x| *x as usize).collect()
+            } else {
+                panic!("Tensor data type must be int64")
+            }
+        }
+        _ => panic!("Only tensor input is valid for shape"),
+    };
+
+    let ends = match &node.inputs[2].ty {
+        ArgType::Tensor(tensor) => {
+            assert_eq!(tensor.dim, 1, "Slice: ends tensor must be 1D");
+            if let Some(Data::Int64s(shape)) = end_value.as_ref() {
+                shape.iter().map(|x| *x as usize).collect()
+            } else {
+                panic!("Tensor data type must be int64")
+            }
+        }
+        _ => panic!("Only tensor input is valid for shape"),
+    };
```

Comment:
According to the [spec](https://github.com/onnx/onnx/blob/main/docs/Operators.md#Slice), we should handle negative values for `starts`, `end` and `axes`.

We should also be able to map the `axes` specified to the correct ranges for `tensor.slice(ranges)`.

Also, we should still parse steps and make sure that values != 1 are not accepted (panic) since Burn only supports regular ranges with step 1.

Reply:
Makes sense, I will put those checks in place

Reply:
How would we handle negative values for `starts`, `ends` and `axes` since we would need to know the length of the input to calculate the new index and that's only known at runtime? I could always throw an error, but there might be a better way to do this.
`a[-1] == a[a.len() - 1]`

Reply:
Oh right. I just checked the implementation for other ops which deal with negative axes and turns out that it's assumed the tensor dims are known when parsing the graph in these nodes... see for example the [shape node config](https://github.com/tracel-ai/burn/blob/main/crates/burn-import/src/onnx/op_configuration.rs#L1007-L1013).

In this case, we have two options:
1) Assume the `tensor.dim` is valid at this point (as is done in the current ops)
2) Throw an error for negative axis.

I don't have any preference per se.

Reply:
Cool, thanks!